api/client/dynamicwindows/dynamicwindows.go

```diff
@@ -0,0 +1,97 @@
+/**
+ * Teleport
+ * Copyright (C) 2024 Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package dynamicwindows
+
+import (
+	"context"
+	dynamicwindows "github.com/gravitational/teleport/api/gen/proto/go/teleport/dynamicwindows/v1"
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/trace"
+)
+
+// Client is a DynamicWindowsDesktop client.
+type Client struct {
+	grpcClient dynamicwindows.DynamicWindowsServiceClient
+}
+
+// NewClient creates a new StaticHostUser client.
+func NewClient(grpcClient dynamicwindows.DynamicWindowsServiceClient) *Client {
+	return &Client{
+		grpcClient: grpcClient,
+	}
+}
+
+func (c *Client) GetDynamicWindowsDesktop(ctx context.Context, name string) (types.DynamicWindowsDesktop, error) {
+	if name == "" {
+		return nil, trace.BadParameter("missing name")
+	}
+	desktop, err := c.grpcClient.GetDynamicWindowsDesktop(ctx, &dynamicwindows.GetDynamicWindowsDesktopRequest{
+		Name: name,
+	})
+	return desktop, trace.Wrap(err)
+}
+
+func (c *Client) ListDynamicWindowsDesktop(ctx context.Context, pageSize int, pageToken string) ([]types.DynamicWindowsDesktop, string, error) {
+	resp, err := c.grpcClient.ListDynamicWindowsDesktops(ctx, &dynamicwindows.ListDynamicWindowsDesktopsRequest{
+		PageSize:  int32(pageSize),
```

Comment:
No action needed (as we do that already in other places), but page size should never be negative.

Reply:
No longer there, as we use generic service now